xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -779,6 +781,28 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
     return;
   }
 
+  if (iWindowID == WINDOW_VIDEO_NAV)
+  {
+    if (params.front().substr(0, 4) != "root" && params.front().substr(0, 10) != "library://")
+    {
+      VECSOURCES& vecSources = *CMediaSourceSettings::GetInstance().GetSources("video");
+
+      if (!g_passwordManager.IsDatabasePathUnlocked(params.front(), vecSources))
+      {
+        // try to find the best matching source
+        bool bName = false;
+        int iIndex = CUtil::GetMatchingSource(params.front(), vecSources, bName);
+        if (iIndex > -1 && iIndex < (int)vecSources.size())
+          if (vecSources[iIndex].m_iHasLock >= 2)
```

Comment:
If you're interested in a follow-up PR, replacing this magic constant with an enum would be a welcome change.

Reply:
and done ;)